### PR TITLE
editoast: migrate endpoint to create a rolling_stock livery

### LINKED ISSUE #3498: migrate POST /rolling-stock/{id}/livery endpoint to editoast
# Description
Migrate `POST /rolling-stock/{id}/livery` endpoint to editoast. This endpoint is used to create a new livery for a given rolling-stock

# Goal
Drop API service.

# Acceptance Criteria
- [ ] the frontend use the editoast endpoint and the user does not see any difference
- [ ] the endpoint doesn't exist anymore in API

### PR DESCRIPTION
**Pull request by clarani**
closes #3498 

:warning: this new endpoint is not tested because TestRequest does not support multipart for the moment. Here is a ticket #3931 to add tests on this specific endpoint (in the future).

To test this endpoint, you can use postman with this configuration (select form-data for the body, use the images in editoast/src/tests if you want). You can add several images for a single request.
image

### CODE REVIEW

**Review by clarani — COMMENTED** [review]
Ready for review

**Review by younesschrifi — APPROVED** [review]
LGTM !! Nice PR :)

File: editoast/src/views/rolling_stocks.rs [hunk 1]
```diff
@@ -95,6 +112,142 @@ async fn create(
...
+
+        let image = ImageReader::new(Cursor::new(buffer))
+            .with_guessed_format()
+            .unwrap();
+
+        let image = match image.decode() {
+            Ok(image) => image,
+            Err(_) => return Err(RollingStockError::CannotReadImage.into()),
+        };
+        max_height = cmp::max(max_height, image.height());
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]:
```suggestion
        max_height = max_height.max(image.height());
```

File: editoast/src/views/rolling_stocks.rs [hunk 2]
```diff
@@ -95,6 +112,142 @@ async fn create(
...
+    db_pool: Data<DbPool>,
+    rolling_stock_id: Path<i64>,
+    MultipartForm(form): MultipartForm<RollingStockLiveryCreateForm>,
+) -> Result<Json<RollingStockLivery>> {
+    let rolling_stock_id = rolling_stock_id.into_inner();
+    let _rolling_stock =
+        match RollingStockModel::retrieve(db_pool.clone(), rolling_stock_id).await? {
+            Some(rolling_stock) => rolling_stock,
+            None => return Err(RollingStockError::NotFound { rolling_stock_id }.into()),
+        };
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]:
```suggestion
    let rolling_stock = RollingStockModel::retrieve(db_pool.clone(), rolling_stock_id).await?;

    if rolling_stock.is_none() {
        return Err(RollingStockError::NotFound { rolling_stock_id }.into());
    }
```

File: editoast/src/views/rolling_stocks.rs [hunk 3]
```diff
@@ -95,6 +112,142 @@ async fn create(
...
+        compound_image_width: total_width,
+        images: separated_images,
+    })
+}
+
+async fn create_compound_image(
+    db_pool: Data<DbPool>,
+    compound_image_height: u32,
+    compound_image_width: u32,
+    images: Vec<DynamicImage>,
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: This function could take a `FormattedImages` instead.

**Review by flomonster — APPROVED** [review]
Thanks for this PR! This looks good to me. I've suggested changes to slightly improve the code.

File: editoast/src/views/rolling_stocks.rs (hunk 1 above)
Comment (clarani) [reply editoast/src/views/rolling_stocks.rs]: Done

File: editoast/src/views/rolling_stocks.rs (hunk 2 above)
Comment (clarani) [reply editoast/src/views/rolling_stocks.rs]: Done

File: editoast/src/views/rolling_stocks.rs (hunk 3 above)
Comment (clarani) [reply editoast/src/views/rolling_stocks.rs]: Done

**Pull request merged**